pytest failing due to old pandas installation and requirements.txt not demanding pandas >= 1.1.0  (v2.0.0)

After pip installing version 2.0.0 from cloned repository the pytest was failing (see below). Upgrading pandas resolved the issue (version 2.0.0 requires pandas >= 1.1.0, but does not specify this requirement.txt). 

Possibly wouldn't usually cause an issue, but I had previously installed using `pip install anesthetic`, which demands pandas==1.0.5.

```
============================================================================================== FAILURES ===============================================================================================
_____________________________________________________________________________________ test_WeightedDataFrame_mean _____________________________________________________________________________________

    def test_WeightedDataFrame_mean():
        df = test_WeightedDataFrame_constructor()
        mean = df.mean()
        assert isinstance(mean, Series)
        assert_allclose(mean, 0.5, atol=1e-2)

        mean = df.mean(axis=1)
>       assert isinstance(mean, WeightedSeries)
E       assert False
E        +  where False = isinstance(#      weights \n0      0.765605    0.622255\n1      0.050536    0.538144\n2      0.997545    0.764341\n3      0.067672   ...0482\n99997  0.831937    0.394225\n99998  0.242556    0.317836\n99999  0.072707    0.511031\nLength: 100000, dtype: float64, WeightedSeries)

tests/test_weighted_pandas.py:86: AssertionError
_____________________________________________________________________________________ test_WeightedDataFrame_std ______________________________________________________________________________________

    def test_WeightedDataFrame_std():
        df = test_WeightedDataFrame_constructor()
        std = df.std()
        assert isinstance(std, Series)
        assert_allclose(std, (1./12)**0.5, atol=1e-2)

        std = df.std(axis=1)
>       assert isinstance(std, WeightedSeries)
E       assert False
E        +  where False = isinstance(#      weights \n0      0.765605    0.084883\n1      0.050536    0.111273\n2      0.997545    0.285251\n3      0.067672   ...2308\n99997  0.831937    0.057368\n99998  0.242556    0.095397\n99999  0.072707    0.152844\nLength: 100000, dtype: float64, WeightedSeries)

tests/test_weighted_pandas.py:97: AssertionError
_____________________________________________________________________________________ test_WeightedDataFrame_sem ______________________________________________________________________________________

    def test_WeightedDataFrame_sem():
        df = test_WeightedDataFrame_constructor()
        sem = df.sem()
        assert isinstance(sem, Series)
        assert_allclose(sem, (1./12)**0.5/np.sqrt(df.neff()), atol=1e-2)

        sem = df.sem(axis=1)
>       assert isinstance(sem, WeightedSeries)
E       assert False
E        +  where False = isinstance(#      weights \n0      0.765605    0.049007\n1      0.050536    0.064243\n2      0.997545    0.164690\n3      0.067672   ...5256\n99997  0.831937    0.033121\n99998  0.242556    0.055077\n99999  0.072707    0.088245\nLength: 100000, dtype: float64, WeightedSeries)

tests/test_weighted_pandas.py:147: AssertionError
___________________________________________________________________________________ test_WeightedDataFrame_kurtosis ___________________________________________________________________________________

    def test_WeightedDataFrame_kurtosis():
        df = test_WeightedDataFrame_constructor()
        kurtosis = df.kurtosis()
        assert isinstance(kurtosis, Series)
        assert_allclose(kurtosis, 9./5, atol=1e-2)
        assert_array_equal(df.kurtosis(), df.kurt())

        kurtosis = df.kurtosis(axis=1)
>       assert isinstance(kurtosis, WeightedSeries)
E       assert False
E        +  where False = isinstance(#      weights \n0      0.765605   NaN\n1      0.050536   NaN\n2      0.997545   NaN\n3      0.067672   NaN\n4      0.20946...\n99996  0.077228   NaN\n99997  0.831937   NaN\n99998  0.242556   NaN\n99999  0.072707   NaN\nLength: 100000, dtype: float64, WeightedSeries)

tests/test_weighted_pandas.py:158: AssertionError
========================================================================================== warnings summary ===========================================================================================
tests/test_plot.py::test_contour_plot_2d[kde_contour_plot_2d]
  /Users/polychord/Code_clones/anesthetic/anesthetic/plot.py:632: UserWarning: No contour levels were found within the data range.
    cont = ax.tricontour(tri, p, contours, zorder=zorder,

tests/test_samples.py::test_weighted_merging
tests/test_samples.py::test_weighted_merging
tests/test_samples.py::test_weighted_merging
tests/test_samples.py::test_weighted_merging
  /usr/local/anaconda3/lib/python3.8/site-packages/numpy/lib/function_base.py:412: RuntimeWarning: invalid value encountered in multiply
    avg = np.multiply(a, wgt, dtype=result_dtype).sum(axis)/scl

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================================================= short test summary info =======================================================================================
FAILED tests/test_weighted_pandas.py::test_WeightedDataFrame_mean - assert False
FAILED tests/test_weighted_pandas.py::test_WeightedDataFrame_std - assert False
FAILED tests/test_weighted_pandas.py::test_WeightedDataFrame_sem - assert False
FAILED tests/test_weighted_pandas.py::test_WeightedDataFrame_kurtosis - assert False
=================================================================== 4 failed, 104 passed, 1 xfailed, 5 warnings in 92.61s (0:01:32) ===================================================================
```

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -7,6 +7,8 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  schedule:
+    - cron: "0 0 * * *"
 
   workflow_dispatch:
 

--- a/README.rst
+++ b/README.rst
@@ -10,8 +10,6 @@ anesthetic: nested sampling visualisation
 .. image:: https://github.com/williamjameshandley/anesthetic/workflows/CI/badge.svg?branch=master
    :target: https://github.com/williamjameshandley/anesthetic/actions?query=workflow%3ACI+branch%3Amaster
    :alt: Build Status
-.. image:: https://circleci.com/gh/williamjameshandley/anesthetic.svg?style=svg
-   :target: https://circleci.com/gh/williamjameshandley/anesthetic
 .. image:: https://codecov.io/gh/williamjameshandley/anesthetic/branch/master/graph/badge.svg
    :target: https://codecov.io/gh/williamjameshandley/anesthetic
    :alt: Test Coverage Status

--- a/README.rst
+++ b/README.rst
@@ -7,8 +7,8 @@ anesthetic: nested sampling visualisation
 :Homepage: https://github.com/williamjameshandley/anesthetic
 :Documentation: http://anesthetic.readthedocs.io/
 
-.. image:: https://travis-ci.com/williamjameshandley/anesthetic.svg?branch=master
-   :target: https://travis-ci.com/williamjameshandley/anesthetic
+.. image:: https://github.com/williamjameshandley/anesthetic/workflows/CI/badge.svg?branch=master
+   :target: https://github.com/williamjameshandley/anesthetic/actions?query=workflow%3ACI+branch%3Amaster
    :alt: Build Status
 .. image:: https://circleci.com/gh/williamjameshandley/anesthetic.svg?style=svg
    :target: https://circleci.com/gh/williamjameshandley/anesthetic

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 scipy>=1.2.0
 numpy
-pandas
+pandas>=1.1.0
 matplotlib>=3.3.0

--- a/requirements_extras.txt
+++ b/requirements_extras.txt
@@ -1,6 +1,6 @@
 scipy>=1.2.0
 numpy
-pandas
+pandas>=1.1.0
 matplotlib>=3.3.0
 fastkde
 astropy

